avocado/plugins/runners/vmimage.py

```diff
@@ -0,0 +1,114 @@
+import sys
+import traceback
+from multiprocessing import set_start_method
+
+from avocado.core.nrunner.app import BaseRunnerApp
+from avocado.core.nrunner.runner import BaseRunner
+from avocado.core.utils import messages
+from avocado.plugins.vmimage import download_image
+from avocado.utils import vmimage
+
+
+class VMImageRunner(BaseRunner):
+    """
+    Runner for dependencies of type vmimage.
+    This runner uses the vmimage plugin's download_image function which handles:
+    1. Checking if the image exists in cache
+    2. Downloading the image if not in cache
+    3. Storing the image in the configured cache directory
+    4. Returning the cached image path
+    """
+
+    name = "vmimage"
+    description = "Runner for dependencies of type vmimage"
+
+    def run(self, runnable):
+        try:
+            yield messages.StartedMessage.get()
+
+            provider = runnable.kwargs.get("provider")
+            version = runnable.kwargs.get("version")
+            arch = runnable.kwargs.get("arch")
+
+            if not all([provider, version, arch]):
+                stderr = "Missing required parameters: provider, version, and arch"
+                yield messages.StderrMessage.get(stderr.encode())
+                yield messages.FinishedMessage.get("error")
+                return
+
+            try:
+                yield messages.StdoutMessage.get(
+                    f"Getting VM image for {provider} {version} {arch}".encode()
+                )
+
+                try:
+                    provider_normalized = provider.lower()
+                    available_providers = [
+                        p.name.lower() for p in vmimage.IMAGE_PROVIDERS
+                    ]
+                    if provider_normalized not in available_providers:
+                        raise ValueError(
+                            f"Provider '{provider}' not found. Available providers: {', '.join(available_providers)}"
+                        )
+
+                    # download_image will use cache if available, otherwise download
+                    image = download_image(provider_normalized, version, arch)
```

Comment:
To get all logs from the vmimage plugin, you need to set up logs before you will call the plugin. You can use `avocado.core.utils.messages.start_logging()` see [avocado-instrumented](https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/runners/avocado_instrumented.py#L95).